### PR TITLE
add support for UDP graphite client

### PR DESCRIPTION
**Pull request by Crim**

### CODE REVIEW

File: src/main/java/com/verisign/storm/metrics/reporters/graphite/GraphiteReporter.java [hunk 1]
```diff
@@ -79,7 +81,14 @@ public GraphiteReporter() {
     graphiteSocketAddr = new InetSocketAddress(graphiteHost, graphitePort);
 
     serverFingerprint = graphiteSocketAddr.getAddress() + ":" + graphiteSocketAddr.getPort();
-    this.graphite = new Graphite(graphiteSocketAddr);
+
+    if (conf.containsKey(GRAPHITE_PROTOCOL_OPTION) && ((String)conf.get(GRAPHITE_PROTOCOL_OPTION)).equalsIgnoreCase("udp")) {
```
Comment (Crim) [inline src/main/java/com/verisign/storm/metrics/reporters/graphite/GraphiteReporter.java]: I wasn't sure if it'd be better to keep backwards compat with existing configs and default to TCP, or to throw an exception if an appropriate config value was not defined

Comment (KevinJMao) [inline src/main/java/com/verisign/storm/metrics/reporters/graphite/GraphiteReporter.java]: Maintaining backwards compatibility is preferable. This should be okay.

File: src/test/java/com/verisign/storm/metrics/reporters/UdpGraphiteReporterTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    reporterConfig.put(GraphiteReporter.GRAPHITE_PORT_OPTION, graphitePort.toString());
+    reporterConfig.put(GraphiteReporter.GRAPHITE_PROTOCOL_OPTION, "udp");
+
+    graphiteReporter = new GraphiteReporter();
+    graphiteReporter.prepare(reporterConfig);
+    graphiteReporter.connect();
+  }
+
+  private void acceptClientConnection() throws IOException {
+    //socketChannel = graphiteServer.accept();
```
Comment (KevinJMao) [inline src/test/java/com/verisign/storm/metrics/reporters/UdpGraphiteReporterTest.java]: Can we remove this method definition if it isn't being used?

File: src/main/java/com/verisign/storm/metrics/reporters/graphite/GraphiteReporter.java (hunk 1 above)
Comment (KevinJMao) [inline src/main/java/com/verisign/storm/metrics/reporters/graphite/GraphiteReporter.java]: LGTM overall. Can you update README.md and CHANGELOG.md with your changes?

**Comment by KevinJMao** [comment]
LGTM. Can you update README.md and CHANGELOG.md with your changes?

File: src/test/java/com/verisign/storm/metrics/reporters/UdpGraphiteReporterTest.java (hunk 2 above)
Comment (Crim) [inline src/test/java/com/verisign/storm/metrics/reporters/UdpGraphiteReporterTest.java]: whoops.  I'll pull that guy out.

**Comment by Crim** [comment]
I updated the changelog and version in gradel to 2.4.  I was unsure of the best place to note the configuration option in the README.md file however.

**Comment by KevinJMao** [comment]
Adding the optional field in the example `storm.yaml` snippet should be fine. Maybe something like

##### Reporting Metrics to Graphite
```yaml
---
### Note: This is Storm's storm.yaml configuration file

# Controls the time interval between metric reports
topology.builtin.metrics.bucket.size.secs: 10
topology.metrics.consumer.register:
  - class: "com.verisign.storm.metrics.GraphiteMetricsConsumer"
    parallelism.hint: 1
    argument:
      metrics.reporter.name: "com.verisign.storm.metrics.reporters.graphite.GraphiteReporter"
      metrics.graphite.host: "graphite.example.com"
      metrics.graphite.port: "2003"
      metrics.graphite.prefix: "storm.test"
      metrics.graphite.min-connect-attempt-interval-secs: "5"
      # Optional arguments can also be supplied to enable UDP
      metrics.graphite.protocol: "udp"
```

**Comment by Crim** [comment]
Done!  Let me know if there are any other tweaks you'd like to see. :)

**Comment by KevinJMao** [comment]
LGTM

**Pull request merged**